### PR TITLE
Respect htmlwidgets.TOJSON_ARGS in server-side processing mode

### LINKED ISSUE #496: display Inf as Inf not like NA
Hi, 

with This code : 

```
data(iris)
iris[1,1] <- NA
iris[2,1] <- Inf
iris
DT::datatable(head(iris))
```

we see that iris[1,1] and iris[2,1] are both blank, I thinks that `Inf` should be notified as `Inf`

I can sure do this kind of trick : 
```
library(dplyr)
iris[is.na(iris)]<-"NA"
DT::datatable(head(iris) %>% mutate_all(as.character))

``` 

But the sorting will be alphabeticaly : 1,10,11...,2,20  (because of character)

I there already a "nice" way to do this ?

### PR DESCRIPTION
**Pull request by shrektan**
closes #496 

### example 

```r
library(shiny)

dat <- data.frame(
  a = c(5, 6, NA, Inf, 10, 100),
  b = c(letters[1:4], NA, NA),
  c = factor(c(letters[1:4], NA, NA)),
  stringsAsFactors = FALSE
)

shinyApp(
  ui = fluidPage(
    radioButtons("na_inf", "na_inf", c("null", "string"), inline = TRUE),
    fluidRow(
      column(
        6,
        h3("client mode"),
        DT::dataTableOutput("client_dt")
      ),
      column(
        6,
        h3("server mode"),
        DT::dataTableOutput("server_dt")
      )
    )
  ),
  server = function(input,output){
    observe(
      options("htmlwidgets.TOJSON_ARGS" = list(na = input$na_inf))
    )
    output$client_dt <- DT::renderDataTable({
      input$na_inf
      DT::datatable(head(dat))
    }, server = FALSE)
    output$server_dt <- DT::renderDataTable({
      input$na_inf
      DT::datatable(head(dat))
    }, server = TRUE)
  },
  options = list(port = 6815)
)
```

### CODE REVIEW

File: R/shiny.R [hunk 1]
```diff
@@ -379,7 +380,9 @@ sessionDataURL = function(session, data, id, filter) {
     res = tryCatch(filter(data, params), error = function(e) {
       list(error = as.character(e))
     })
-    httpResponse(200, 'application/json', enc2utf8(toJSON(res, dataframe = 'rows')))
+
+    jsonArgs = c(list(x = res, dataframe = 'rows'), attr(data, "TOJSON_ARGS", exact = TRUE))
+    httpResponse(200, 'application/json', enc2utf8(do.call(toJSON, jsonArgs)))
```
Comment (yihui) [inline R/shiny.R]: I only want to keep these two lines in this PR, and change `attr(data, "TOJSON_ARGS", exact = TRUE)` to `getOption('htmlwidgets.TOJSON_ARGS')`, i.e. only support the global option `htmlwidgets.TOJSON_ARGS`.

Comment (shrektan) [reply R/shiny.R]: Yes, I feel the same, too. How about an option in DT? I’m afraid if the user uses another HTML widgets, the global option may lead to undesirable results on the other widget. What’s your option?

Comment (yihui) [reply R/shiny.R]: Let's consider that in the future when users complain about it :) I don't think many people actually know this option at all. In my eyes, it is a quite trivial matter.

**Comment by shrektan** [comment]
Understand. I've force pushed the branch as your suggested and the example above gets updated as well. 😃

**Comment by geoabi** [comment]
Is there any way to use this solution for only one table or without affecting other widgets, like Plotly? My current solution is to use  `options(htmlwidgets.TOJSON_ARGS = list(na = 'string'))` before render a table and to use `options(htmlwidgets.TOJSON_ARGS = NULL)`  before render a plot with Plotly. 

Maybe an extra argument to `datatable()` could be a more elegant solution. Thank you.

**Comment by vnijs** [comment]
I just realized that @geoabi meant that setting `options(htmlwidgets.TOJSON_ARGS = list(na = 'string'))` actually causes an error when using `plotly`. See example below:

```r
Error in jsonlite::toJSON(x, digits = 50, auto_unbox = TRUE, force = TRUE,  : 
  formal argument "na" matched by multiple actual arguments
```

```r
options(htmlwidgets.TOJSON_ARGS = list(na = "string"))

library(plotly)
dat <- data.frame(
  time = factor(c("Lunch","Dinner"), levels=c("Lunch","Dinner")),
  total_bill = c(14.89, 17.23)
)

p <- ggplot(data=dat, aes(x=time, y=total_bill)) +
  geom_bar(stat="identity")

ggplotly(p)

**Comment by shrektan** [comment]
I see. However, I agree with @yihui that the parameter number has already  been a lot in `DT::datatable()`. An alternative approach is to provide a new option `DT.TOJSON_ARGS`. Anyway, it will be better if you can open a new issue. Thanks.

**Comment by yihui** [comment]
`DT.TOJSON_ARGS` sounds like a good idea to me.

**Pull request merged**